### ISSUE TITLE
Improve `libtofs.jinja` to handle custom grains that are lists, etc.

### ISSUE THREAD

**Issue by myii** [saltstack-formulas/template-formula#149]
While debugging TOFS on Slack (see https://freenode.logbot.info/saltstack-formulas/20190722#c2428921), found that a custom grains for `role` (using a list) crashes out `libtofs.jinja`.

```salt
    Data failed to compile:
----------
    Rendering SLS 'base:ssf.config.file' failed: Jinja variable 'list object' has no attribute 'strip'
/var/cache/salt/minion/files/base/ssf/libtofs.jinja(90):
---
[...]
        {%- if fs %}
          {%- set fs_dir = salt['config.get'](fs, fs) %}
        {%- else %}
          {%- set fs_dir = salt['config.get'](tplroot ~ ':tofs:dirs:default', 'default') %}
        {%- endif %}
        {%- set url = [    <======================
            '- salt:/',
            path_prefix_inc_ext.strip('/'),
            files_dir.strip('/'),
            fs_dir.strip('/'),
            src_file.strip('/'),
[...]
```

Need to improve the library to deal with these cases as well.

Fancy taking this on @baby-gnu?  Or shall I work on it?

**Comment by myii** [comment]
Another point that came up in that discussion was that both the documentation and the comments in `libtofs.jinja` indicate that the overrides should be under an `/etc` directory!  So we need to look at making these defaults clearer, or even having a custom TOFS document for each formula... with the `ssf-formula` in development, this will become fairly easy to maintain (but not pleasant to actually work on).

**Comment by baby-gnu** [comment]
@myii I'm not sure I understand the stuff with the override under  `/etc/`. 

Do you refer to the way the author reproduce the destination path under `files/`? Like:

```
/srv/saltstack/salt-formulas/ntp-saltstack-formula/
  ntp/
    files/
      default/
        etc/
          ntp.conf.jinja
```

If the destination path is `/var/lib/template/foo.txt` then the author would have used `files/default/var/lib/template/foo.txt.jinja`.

This looks to be an author's habit and not something to be enforced. It could be misunderstood but I think it's a way for the author to promote his way of doing it.

**Comment by myii** [comment]
@baby-gnu Don't worry about the`/etc` issue too much, I already covered it in the discussion and will work out how the docs/comments can be improved so that they don't cause confusion.

**Comment by myii** [comment]
Fixed in #150.
